test/bdd/pkg/issuecredential/issuecredential_sdk_steps.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package issuecredential
+
+import (
+	"errors"
+	"time"
+
+	"github.com/cucumber/godog"
+
+	"github.com/hyperledger/aries-framework-go/pkg/client/didexchange"
+	"github.com/hyperledger/aries-framework-go/pkg/client/issuecredential"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/decorator"
+	protocol "github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/issuecredential"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
+	verifiableStore "github.com/hyperledger/aries-framework-go/pkg/store/verifiable"
+	"github.com/hyperledger/aries-framework-go/test/bdd/pkg/context"
+)
+
+const retryAttempts = 3
+
+func getVCredential() *verifiable.Credential {
+	const referenceNumber = 83294847
+
+	var issued = time.Date(2010, time.January, 1, 19, 23, 24, 0, time.UTC)
+
+	return &verifiable.Credential{
+		Context: []string{
+			"https://www.w3.org/2018/credentials/v1",
+			"https://www.w3.org/2018/credentials/examples/v1"},
+		ID: "http://example.edu/credentials/1872",
+		Types: []string{
+			"VerifiableCredential",
+			"UniversityDegreeCredential"},
+		Subject: struct {
+			ID string
+		}{ID: "SubjectID"},
+		Issuer: verifiable.Issuer{
+			ID:   "did:example:76e12ec712ebc6f1c221ebfeb1f",
+			Name: "Example University",
+		},
+		Issued:  &issued,
+		Schemas: []verifiable.TypedID{},
+		CustomFields: map[string]interface{}{
+			"referenceNumber": referenceNumber,
+		},
+	}
+}
+
+// SDKSteps is steps for the issuecredential using client SDK
+type SDKSteps struct {
+	bddContext *context.BDDContext
+	clients    map[string]*issuecredential.Client
+	actions    map[string]chan service.DIDCommAction
+	events     map[string]chan service.StateMsg
+}
+
+// NewIssueCredentialSDKSteps creates steps for the issuecredential with SDK
+func NewIssueCredentialSDKSteps(ctx *context.BDDContext) *SDKSteps {
+	return &SDKSteps{
+		bddContext: ctx,
+		clients:    make(map[string]*issuecredential.Client),
+		actions:    make(map[string]chan service.DIDCommAction),
+		events:     make(map[string]chan service.StateMsg),
+	}
+}
+
+// RegisterSteps registers agent steps
+func (a *SDKSteps) RegisterSteps(s *godog.Suite) {
+	s.Step(`^"([^"]*)" sends requests credential to the "([^"]*)"$`, a.sendsRequest)
+	s.Step(`^"([^"]*)" accepts request and sends credential to the Holder$`, a.acceptRequest)
+	s.Step(`^"([^"]*)" accepts credential with name "([^"]*)"$`, a.acceptCredential)
+	s.Step(`^"([^"]*)" checks that credential is being stored under "([^"]*)" name$`, a.checkCredential)
+}
+
+func (a *SDKSteps) checkCredential(agent, name string) error {
+	return retry(retryAttempts, time.Second, func() error {
```

Comment:
Why do we need retry here ?

Reply:
To wait until the protocol saves payload.

Reply:
Can we wait for message event from Continue and then invoke the get VC ?

Reply:
Done!